### PR TITLE
Create Interfaces

### LINKED ISSUE #2: Create interfaces
write:
- model interface
- viewmodel interface
- message interface
- validation interface
- repository interface

dependencies: #1

### PR DESCRIPTION
**Pull request by levkovalenko**
- [x] model interface
- [x] viewmodel interface
- [x] message interface
- [x] validation interface
- [x] repository interface
- [x] converter interface
close #2

### CODE REVIEW

File: Domain/basicModelInterfaces.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package Domain
+
+type UserModelInterface interface {
+	GetFirstName() string
+	GetLastName() string
+	GetBirthDate() string
```
Comment (malinink) [inline Domain/basicModelInterfaces.go:6]: In my opinion in domain model bidthDate should be type of date, sex - enum(or int, that could be stored as const in userModel class), material status - same as sex.

File: Domain/applicationModelInterface.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package Domain
+
+type UserApplicationModelInterface interface {
+	GetId() int
+	GetPassport() PassportApplicationModelInterface
+	GetJobs() []JobsApplicationModelInterface
+	UserModelInterface
```
Comment (malinink) [inline Domain/applicationModelInterface.go:7]: It is a bad idea to inherit same interface as domain userModel does.
I prefer one another interface, placed in application. Same approach is correct for infrastructure layer also.

File: Domain/converterInterface.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package Domain
+
+type ConverterInterface interface {
+	FromViewModel(model UserViewModelInterface) UserApplicationModelInterface
```
Comment (malinink) [inline Domain/converterInterface.go:4]: look something like https://github.com/devfeel/mapper  to solve that problem.

**Review by malinink — CHANGES_REQUESTED** [review]

File: Domain/applicationModelInterface.go (hunk 2 above)
Comment (levkovalenko) [reply Domain/applicationModelInterface.go:7]: UserModelInterface isn't a Domain Model, It  is the same part of Interface for application model and view model. Actually may be it's name isn't suitable and I should change it to UserBasicModelInterface

File: Domain/basicModelInterfaces.go (hunk 1 above)
Comment (levkovalenko) [reply Domain/basicModelInterfaces.go:6]: @malinink actually, UserModelInterface isn't a Domain Mode

File: Domain/converterInterface.go (hunk 3 above)
Comment (levkovalenko) [reply Domain/converterInterface.go:4]: Oh, it is interesting solution, thanks

File: Domain/basicModelInterfaces.go (hunk 1 above)
Comment (malinink) [reply Domain/basicModelInterfaces.go:6]: It is not clear to me

**Review by malinink — APPROVED** [review]

**Comment by levkovalenko** [comment]
@malinink i use this interface cause it contains the similar methods of other interfaces. it is actually abstract interface, where won't be any implementation

**Pull request merged**